varcode/core_logic.py

```diff
@@ -381,6 +383,78 @@ def infer_exonic_effect(variant, transcript):
         variant,
         transcript)
 
+def find_nearest_exon(start, end, exons):
+    """
+    Finds nearest exon to an intronic variant, returns the
+    distance to that exon, along with the Exon object itself.
+    """
+    best_distance = float("inf")
+    best_exon = None
+    for exon in exons:
+        if exon.start > end:
+            # distance from end of variant to start of exon
+            # (when variant starts before the exon)
+            distance = exon.start - end
+        elif exon.end < start:
+            distance = start - exon.end
+        else:
+            assert exon.start <= end and exon.end >= start, \
+                "Expected interval %d:%d to overlap with exon %s" % (
+                    start, end, exon)
+            return 0, exon
+
+        if best_distance > distance:
+            best_distance = distance
+            best_exon = exon
+
+    return best_exon, best_exon
+
+
+def infer_intronic_effect(variant, transcript, nearest_exon, distance_to_exon):
```

Comment:
nit: I'd consider not passing around the distance_to_exon value. it's cheap to compute given the variant and the exon, and passing it around opens an opportunity for inconsistency (would want to have a helper function to do this though)